### PR TITLE
Add docs for Microsoft Teams

### PR DESCRIPTION
**Pull request by SomtochiAma**
Adds documentation for setting up notification for Microsoft teams.

Fix: https://github.com/fluxcd/website/issues/738

Signed-off-by: Somtochi Onyekwere <somtochionyekwere@gmail.com>

### CODE REVIEW

File: docs/spec/v1beta1/provider.md [hunk 1]
```diff
@@ -414,6 +414,29 @@ spec:
   secretRef:
     name: slack-token
 ```
+### MS Teams
+
+On the Microsoft Teams UI,
+- Open the settings of the channel you want the notifications to be sent to.
+- Click on `Connectors`.
+- Click on the `Add` button for Incoming Webhook.
+- Click on Configure and copy the webhook url given.
```
Comment (makkes) [inline docs/spec/v1beta1/provider.md:423]: nit: this might better be a numbered list.

File: docs/spec/v1beta1/provider.md [hunk 2]
```diff
@@ -414,6 +414,29 @@ spec:
...
+See documentation [here](https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook).
```
Comment (makkes) [inline docs/spec/v1beta1/provider.md:424]:
```suggestion
See the [documentation of MS Teams Incoming Webhooks](https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook) for further information.
```

**Review by makkes — COMMENTED** [review]
just two nits

**Review by kingdonb — APPROVED** [review]
LGTM 👍 Thank you for taking care of this!

File: docs/spec/v1beta1/provider.md [hunk 3]
```diff
@@ -414,6 +414,29 @@ spec:
   secretRef:
     name: slack-token
 ```
+### MS Teams
+
+On the Microsoft Teams UI,
```
Comment (stefanprodan) [inline docs/spec/v1beta1/provider.md:419]: Please add a new line after this, so it will render nicely on fluxcd.io

**Review by stefanprodan — APPROVED** [review]

**Comment by mloskot** [comment]
Has this content been actually released anywhere?
I'm not finding it on anywhere on the https://fluxcd.io/flux/ - searching for word `connectors` brings just this https://fluxcd.io/flux/components/notification/provider/#microsoft-teams

**Comment by somtochiama** [comment]
Hey @mloskot 

The doc in the PR has been replaced by one from a more recent API version, which comes up in the search. I can see that steps to create a webhook from the UI are no longer there and I will update the new docs to include them. Are there any other parts that you think are missing?

File: docs/spec/v1beta1/provider.md [hunk 4]
```diff
@@ -215,6 +215,70 @@ kubectl create secret generic tls-certs \
...
+
+### MS Teams
+
+Create an incoming webhook on the Microsoft Teams UI:
+
+1. Open the settings of the channel you want the notifications to be sent to.
+2. Click on `Connectors`.
+3. Click on the `Add` button for Incoming Webhook.
+4. Click on Configure and copy the webhook url given.
+
```
Comment (mloskot) [inline docs/spec/v1beta1/provider.md:263]: @somtochiama 
>  Are there any other parts that you think are missing?

Right, so this step-by-step was replaced with links in [the new doc](https://fluxcd.io/flux/components/notification/provider/#microsoft-teams-example):

> create a Secret with [the address](https://fluxcd.io/flux/components/notification/provider/#address-example) set to the [webhook URL](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook#create-incoming-webhooks-1)

The only suggestion I could have is to keep the step by step form as the current single sentence "To configure a..." is not quite explicit, especially to newbies like myself. So, I would go for something like this:

1. Create an [incoming webhook](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?tabs=dotnet#create-incoming-webhooks-1) on the Microsoft Teams and copy the webhook URL.
2. Create a Secret with [the address](https://fluxcd.io/flux/components/notification/provider/#address-example) set to the webhook URL.
3. Create a Provider for Microsoft Teams with a [Secret reference](https://fluxcd.io/flux/components/notification/provider/#address-example):
    ```
    ---
    apiVersion: notification.toolkit.fluxcd.io/v1beta2
    kind: Provider
    ...
    ```

**Pull request merged**